### PR TITLE
fix: deactivate tooltips on open menus

### PR DESCRIPTION
**Pull request by dennyscode**
Ticket: https://lifi.atlassian.net/browse/LF-5174

[screen-capture-no-tooltip-on-open-menu.webm](https://github.com/lifinance/jumper.exchange/assets/43956540/62d8749b-9f95-475e-881e-34245c79932d)

### CODE REVIEW

File: packages/dapp/src/hooks/usePopperIsOpened.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  }, [
+    openMainNavbarMenu,
+    openNavbarChainsMenu,
+    openNavbarSubMenu,
+    openNavbarWalletMenu,
+    openNavbarWalletSelectMenu,
+  ]);
+
+  return popperOpened;
+};
```
Comment (dennyscode) [inline packages/dapp/src/hooks/usePopperIsOpened.ts:37]: I have used a hook for listening on state changes of our menus. 
@chybisov @Abhikumar98 @0xsebn is this too much or good to go? Wdyt?

File: packages/dapp/src/components/ThemeSwitch/ThemeSwitch.tsx [hunk 2]
```diff
@@ -43,6 +44,7 @@ export const ThemeSwitch = () => {
 
   return (
     <Tooltip
+      disableHoverListener={popperOpened}
```
Comment (dennyscode) [inline packages/dapp/src/components/ThemeSwitch/ThemeSwitch.tsx:47]: So many options! `disableHoverListener` works for the most part, but honestly I am not sure if any of those ones may be needed as well. Wdyt? @chybisov @abhishek-lifi @0xsebn 

```
      disableInteractive={popperOpened}
      disableFocusListener={popperOpened}
      disableTouchListener={popperOpened}
```

File: packages/dapp/src/hooks/usePopperIsOpened.ts (hunk 1 above)
Comment (Abhikumar98) [reply packages/dapp/src/hooks/usePopperIsOpened.ts:37]: LGTM

a minor suggestion
```ts
useEffect(() => {
    const isPopperOpened = openMainNavbarMenu ||
        openNavbarWalletMenu ||
        openNavbarSubMenu !== 'None' ||
        openNavbarChainsMenu ||
        openNavbarWalletSelectMenu;

    setPopperOpened(isPopperOpened);
}, [ ... ])
};
```

File: packages/dapp/src/components/ThemeSwitch/ThemeSwitch.tsx (hunk 2 above)
Comment (Abhikumar98) [reply packages/dapp/src/components/ThemeSwitch/ThemeSwitch.tsx:47]: [disableHoverListener](https://mui.com/material-ui/api/tooltip/#Tooltip-prop-disableHoverListener) seems like it should cover all cases.

I guessing this shouldn't be a valid case in phones anyways.

so, LGTM.

File: packages/dapp/src/hooks/usePopperIsOpened.ts (hunk 1 above)
Comment (chybisov) [reply packages/dapp/src/hooks/usePopperIsOpened.ts:37]: @dennyscode this is too much and can be optimized. I suggest ->

```ts
export const usePopperIsOpened = () => {
  const [
    openMainNavbarMenu,
    openNavbarWalletMenu,
    openNavbarSubMenu,
    openNavbarChainsMenu,
    openNavbarWalletSelectMenu,
  ] = useMenuStore((state) => [
    state.openMainNavbarMenu,
    state.openNavbarWalletMenu,
    state.openNavbarSubMenu,
    state.openNavbarChainsMenu,
    state.openNavbarWalletSelectMenu,
  ]);

  return openMainNavbarMenu ||
        openNavbarWalletMenu ||
        openNavbarSubMenu !== 'None' ||
        openNavbarChainsMenu ||
        openNavbarWalletSelectMenu;
};
```

This hook does exactly the same and has -1 re-render.

**Review by chybisov — CHANGES_REQUESTED** [review]
See `usePopperIsOpened` comment

File: packages/dapp/src/hooks/usePopperIsOpened.ts (hunk 1 above)
Comment (dennyscode) [reply packages/dapp/src/hooks/usePopperIsOpened.ts]: Great feedback! Thank you guys

File: packages/dapp/src/components/ThemeSwitch/ThemeSwitch.tsx (hunk 2 above)
Comment (dennyscode) [reply packages/dapp/src/components/ThemeSwitch/ThemeSwitch.tsx:47]: Perfect!

**Review by chybisov — APPROVED** [review]

**Review by Abhikumar98 — APPROVED** [review]

**Pull request merged**